org.eclipse.m2e.core/src/org/eclipse/m2e/core/internal/project/registry/MavenProjectFacade.java

```diff
@@ -59,6 +74,12 @@
 
   public static final String PROP_CONFIGURATORS = MavenProjectFacade.class.getName() + "/configurators";
 
+  private static final String COMPONENTS_PATH = "META-INF/plexus/components.xml";
```

Comment:
I'm annoyed I didn't find constant in Plexus or Maven for that...

Reply:
I even don't found the place where it is parsed ...